httpclient5/src/main/java/org/apache/hc/client5/http/cookie/Cookie.java

```diff
@@ -77,9 +78,26 @@
      * considered immutable. Changing it (e.g. using setTime()) could result
      * in undefined behaviour. Do so at your peril. </p>
      * @return Expiration {@link Date}, or {@code null}.
+     * @deprecated Use {{@link #getExpiryInstant()}}
      */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
     Date getExpiryDate();
```

Comment:
@arturobernalg This looks wrong. Why are you suppressing deprecated method itself and not its usage where required for backward compatibility? Please correct it.

Reply:
Hi @ok2c 
The annotations @SuppressWarnings support method.  through this ways no need annotated where it is used.

Reply:
> The annotations @SuppressWarnings support method. through this ways no need annotated where it is used.

@arturobernalg This is not a matter of taste or preference. Consumers of the library must be warned about API deprecation. The _only_ exception is the internal use of deprecated methods required for backward compatibility.

Reply:
Changed.
ty

Reply:
@arturobernalg I still see a lot of `@SuppressWarnings("DeprecatedIsStillUsed")` over deprecated methods. This is wrong. Please suppress the warning where the deprecated is used for backward compatibility, not the declaration itself.

Reply:
Solved.